src/targets/awsLambdaLayer.ts

```diff
@@ -143,56 +143,145 @@ export class AwsLambdaLayerTarget extends BaseTarget {
+    ): void => {
+      const latestVersionPath = path.posix.join(directory, 'latest.json');
+      if (fs.existsSync(latestVersionPath)) {
+        const previousVersion = fs
+          .readlinkSync(latestVersionPath)
+          .split('.json')[0];
+        createSymlinks(versionFilepath, version, previousVersion);
+      } else {
+        // When no previous versions are found, just create symlinks.
+        createSymlinks(versionFilepath, version);
+      }
     };
 
-    await Promise.all(awsRegions.map(publishRegion));
-  }
+    logger.debug('Fetching AWS regions...');
+    const awsRegions = extractRegionNames(await getRegionsFromAws());
 
-  /**
-   * Publishes the layer to AWS Lambda with the given layer data.
-   * It must contain the buffer for the ZIP archive and the layer name.
-   * Each time you publish with the same layer name, a new version is created.
-   * @param lambda The lambda service object.
-   * @param layerData Details of the layer to be created.
-   */
-  public publishAwsLayer(
-    lambda: Lambda,
-    layerData: PublishLayerVersionRequest
-  ): Promise<PublishLayerVersionCommandOutput> {
-    return lambda.publishLayerVersion(layerData);
-  }
+    /** Publishes new AWS Lambda layers for every runtime. */
+    const publishRuntimes = async (directory: string): Promise<void> => {
+      await Promise.all(
+        this.config.compatibleRuntimes.map(
+          async (runtime: CompatibleRuntime) => {
+            const layerManager = new AwsLambdaLayerManager(
+              runtime,
+              this.config.layerName,
+              this.config.license,
+              artifactBuffer,
+              awsRegions
+            );
 
-  /**
-   * Adds to a layer usage permissions to other accounts.
-   * @param lambda The lambda service object.
-   * @param layerPermissionData Details of the layer and permissions to be set.
-   */
-  public addAwsLayerPermissions(
-    lambda: Lambda,
-    layerPermissionData: AddLayerVersionPermissionRequest
-  ): Promise<any> {
-    return lambda.addLayerVersionPermission(layerPermissionData);
+            let publishedLayers = [];
+            try {
+              publishedLayers = await layerManager.publishToAllRegions();
+            } catch (error) {
+              logger.error(
+                `Did not publish layers for ${runtime.name}. ` +
+                  `Something went wrong with AWS: ${error.message}`
+              );
+              return;
+            }
+
+            // If no layers have been created, don't do extra work updating files.
+            if (publishedLayers.length == 0) {
+              logger.info(`${runtime.name}: no layers published.`);
+              return;
+            } else {
+              logger.info(
+                `${runtime.name}: ${publishedLayers.length} layers published.`
+              );
+            }
+
+            // Base directory for the layer files of the current runtime.
+            const RUNTIME_BASE_DIR = path.posix.join(
```

Comment:
```suggestion
            const runtimeBaseDir = path.posix.join(
```
Let's reserve uppercase identifiers for global/class constants